### PR TITLE
Add simple wm8994 codec driver

### PR DESCRIPTION
**Pull request by ChaochaoCui**

### CODE REVIEW

File: arch/arm/src/stm32f7/stm32f74xx75xx_rcc.c [hunk 1]
```diff
@@ -1005,3 +1006,44 @@ static inline void rcc_enableperipherals(void)
...
+ * Input Parameters:
+ *   msg - Message to print before the register data
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_DEBUG_INFO
+void stm32_rcc_dump_regs(const char *msg)
```
Comment (xiaoxiang781216) [inline arch/arm/src/stm32f7/stm32f74xx75xx_rcc.c:1025]: @ChaochaoCui nobody call stm32_rcc_dump_regs, why not directly remove it?

File: boards/arm/stm32f7/stm32f746g-disco/include/board.h [hunk 2]
```diff
@@ -513,4 +521,31 @@
...
+
+#define DMACHAN_SAI2_A                  DMAMAP_SAI2_A
+#define DMACHAN_SAI2_B                  DMAMAP_SAI2_B
+
+#else
+
+#define GPIO_SAI1_SD_B                  GPIO_SAI1_SD_B_1
+#define DMACHAN_SAI1_B                  DMAMAP_SAI1_B
+
+#if 0
```
Comment (xiaoxiang781216) [inline boards/arm/stm32f7/stm32f746g-disco/include/board.h:540]: why skip the following defintion?

Comment (ChaochaoCui) [reply boards/arm/stm32f7/stm32f746g-disco/include/board.h:540]: for debug. I will remove it.

File: arch/arm/src/stm32f7/stm32f74xx75xx_rcc.c (hunk 1 above)
Comment (ChaochaoCui) [reply arch/arm/src/stm32f7/stm32f74xx75xx_rcc.c:1025]: for debug. I will remove it.

File: boards/arm/stm32f7/stm32f746g-disco/include/board.h (hunk 2 above)
Comment (ChaochaoCui) [reply boards/arm/stm32f7/stm32f746g-disco/include/board.h]: I have removed it.

File: arch/arm/src/stm32f7/hardware/stm32f74xx75xx_pinmap.h [hunk 3]
```diff
@@ -545,12 +545,12 @@
 #define GPIO_I2C2_SMBA_2      (GPIO_ALT|GPIO_AF4|GPIO_SPEED_50MHz|GPIO_PUSHPULL|GPIO_PORTF|GPIO_PIN2)
 #define GPIO_I2C2_SMBA_3      (GPIO_ALT|GPIO_AF4|GPIO_SPEED_50MHz|GPIO_PUSHPULL|GPIO_PORTH|GPIO_PIN6)
 
-#define GPIO_I2C3_SCL_1       (GPIO_ALT|GPIO_AF4|GPIO_SPEED_50MHz|GPIO_OPENDRAIN|GPIO_PORTA|GPIO_PIN8)
-#define GPIO_I2C3_SCL_2       (GPIO_ALT|GPIO_AF4|GPIO_SPEED_50MHz|GPIO_OPENDRAIN|GPIO_PORTH|GPIO_PIN7)
-#define GPIO_I2C3_SDA_1       (GPIO_ALT|GPIO_AF4|GPIO_SPEED_50MHz|GPIO_OPENDRAIN|GPIO_PORTC|GPIO_PIN9)
-#define GPIO_I2C3_SDA_2       (GPIO_ALT|GPIO_AF4|GPIO_SPEED_50MHz|GPIO_OPENDRAIN|GPIO_PORTH|GPIO_PIN8)
-#define GPIO_I2C3_SMBA_1      (GPIO_ALT|GPIO_AF4|GPIO_SPEED_50MHz|GPIO_PUSHPULL|GPIO_PORTA|GPIO_PIN9)
-#define GPIO_I2C3_SMBA_2      (GPIO_ALT|GPIO_AF4|GPIO_SPEED_50MHz|GPIO_PUSHPULL|GPIO_PORTH|GPIO_PIN9)
+#define GPIO_I2C3_SCL_1       (GPIO_ALT|GPIO_AF4|GPIO_SPEED_100MHz|GPIO_OPENDRAIN|GPIO_PORTA|GPIO_PIN8)
```
Comment (davids5) [inline arch/arm/src/stm32f7/hardware/stm32f74xx75xx_pinmap.h:548]: Why this change? We need to really remove the slew rate control and have the boards set it.

File: boards/arm/stm32f7/stm32f746g-disco/include/board.h [hunk 4]
```diff
@@ -366,7 +374,7 @@
 
 #define GPIO_TP_INT  (GPIO_INPUT|GPIO_FLOAT|GPIO_EXTI|GPIO_PORTI|GPIO_PIN13)
 
-#define FT5x06_I2C_ADDRESS          0x38
+#define FT5x06_I2C_ADDRESS          (0x70 >> 1)
```
Comment (davids5) [inline boards/arm/stm32f7/stm32f746g-disco/include/board.h:377]: It is better to list the un shifted address. Is this how it is listed in the datasheet?.

**Review by davids5 — CHANGES_REQUESTED** [review]
Please see questions

File: boards/arm/stm32f7/stm32f746g-disco/include/board.h (hunk 4 above)
Comment (ChaochaoCui) [reply boards/arm/stm32f7/stm32f746g-disco/include/board.h:377]: I see the address in stm32f746g-disco board sch. https://www.st.com/content/ccc/resource/technical/layouts_and_diagrams/schematic_pack/group1/ff/cd/ce/2d/f8/fb/40/69/mb1191-F746NGH6-C01_schematic/files/mb1191-F746NGH6-C01_schematic.pdf/jcr:content/tranAslations/en.mb1191-F746NGH6-C01_schematic.pdf

File: arch/arm/src/stm32f7/hardware/stm32f74xx75xx_pinmap.h (hunk 3 above)
Comment (ChaochaoCui) [reply arch/arm/src/stm32f7/hardware/stm32f74xx75xx_pinmap.h:548]: When I debug wm8994 codec driver,  the STM32F746 communicate with WM8996 through I2C, use 100MHz ok, but 50MHz not ok.

**Comment by xiaoxiang781216** [comment]
@davids5 does @ChaochaoCui 's reply resolve your concern?

**Comment by davids5** [comment]
In the changed to the header file stm32f74xx75xx_pinmap.h. No. We can not change all the boards default I2C settings. It will may break deployed systems. Or worse violate FCC. 

The values need to be adjusted in the board config:

in board.h add
```
define ADJ_SLEW_RATE(p) (((p) & ~GPIO_SPEED_MASK) | (GPIO_SPEED_100MHz))

#define GPIO_I2C3_SCL  ADJ_SLEW_RATE(GPIO_I2C3_SCL_1 ) 
...
```

File: boards/arm/stm32f7/stm32f746g-disco/include/board.h (hunk 4 above)
Comment (davids5) [reply boards/arm/stm32f7/stm32f746g-disco/include/board.h:377]: Defining them shifted,  (what you would see on the bus) then shifting them back is confusing.  I would avoid the confusion, and use 7-bit address in the header file and put a comment that says `/* 7 bit address */`

After all what will `i2cdetect` show?  0x38,  If I then grep 0x38. I will not find this file!

Comment (xiaoxiang781216) [reply boards/arm/stm32f7/stm32f746g-disco/include/board.h:377]: @ChaochaoCui this is minor change, let's keep it as before.

**Comment by xiaoxiang781216** [comment]
Ok, @ChaochaoCui it's better to move the change in arch/arm/src/stm32f7/hardware/stm32f74xx75xx_pinmap.h to another PR, so we can merge this PR first.

**Comment by ChaochaoCui** [comment]
Ok. I will do it.

**Comment by ChaochaoCui** [comment]
@davids5 @xiaoxiang781216 I have restored it. Please review it.

**Comment by davids5** [comment]
Thank you @ChaochaoCui - please fix the nxstyle issues and then squash the commits and force push.
